pyscriptjs/src/components/pyscript.ts

```diff
@@ -18,17 +18,14 @@ export function make_PyScript(runtime: Runtime) {
         async getPySrc(): Promise<string> {
             if (this.hasAttribute('src')) {
                 const url = this.getAttribute('src');
-                const response = await fetch(url);
-                if (response.status !== 200) {
-                    const errorMessage = (
-                        `Failed to fetch '${url}' - Reason: ` +
-                        `${response.status} ${response.statusText}`
-                    );
-                    _createAlertBanner(errorMessage);
+                try {
+                    const response = await fetchIt(url);
+                    return await response.text();
+                } catch(e) {
+                    _createAlertBanner(e.message);
                     this.innerHTML = '';
-                    throw new FetchError(errorMessage);
+                    throw e
```

Comment:
[follow-up]
Another `_createAlertBanner`: this is a code smell ;).
I think that the correct way to handle this case is to raise `UserError` from here, so that it will be caught by the general UserError logic.

We could go even further, by choosing one of these two options:
1. rename it as `FetchUserError` and make it a subclass of `UserError`: this way, *all* fetch errors will be displayed to the user in a proper way. I'm not 100% sure that this is always the case though: maybe there are cases in which a failed fetch is not an user error but an internal error?

2. in case we decide that some are user errors and some are internal errors, we can probably add an option to `fetchIt` to indicate which exception to raise in each case

Reply:
That's a good point, I like the both ideas 😄 
When I was creating the error codes, I was wondering how often will people see weird error codes like 500's. Maybe we could always assume it's a user error? 🤔

Reply:
do you mean HTTP 500? I think that yes, in general we should have a way to inform the user that something went wrong. But it seems like a topic for a broader discussion, let's not block this PR on this